ibm/data_source_ibm_is_bare_metal_server_initialization.go

```diff
@@ -0,0 +1,235 @@
+				isBareMetalServerUserAccountUserName: *acc.Username,
+			}
+			currAccount[isBareMetalServerUserAccountResourceType] = *acc.ResourceType
+			currAccount[isBareMetalServerUserAccountEncryptionKey] = *acc.EncryptionKey.CRN
+			encPassword := base64.StdEncoding.EncodeToString(*acc.EncryptedPassword)
+			currAccount[isBareMetalServerUserAccountEncryptedPassword] = encPassword
+
+			var rsaKey *rsa.PrivateKey
+			// image := *initialization.Image.Name
+			// res := strings.Contains(image, "windows")
+			// if res {
+			if privatekey, ok := d.GetOk(isBareMetalServerPEM); ok {
+				keyFlag := privatekey.(string)
+				keybytes := []byte(keyFlag)
+
+				if keyFlag != "" {
+					block, err := pem.Decode(keybytes)
+					if block == nil {
+						return diag.FromErr(fmt.Errorf("Failed to load the private key from the given key contents. Instead of the key file path, please make sure the private key is pem format"))
+					}
+					isEncrypted := false
+					switch block.Type {
+					case "RSA PRIVATE KEY":
+						isEncrypted = x509.IsEncryptedPEMBlock(block)
+					case "OPENSSH PRIVATE KEY":
+						var err error
+						isEncrypted, err = isOpenSSHPrivKeyEncrypted(block.Bytes)
+						if err != nil {
+							return diag.FromErr(fmt.Errorf("Failed to check if the provided open ssh key is encrypted or not %s", err))
+						}
+					default:
+						return diag.FromErr(fmt.Errorf("PEM and OpenSSH private key formats with RSA key type are supported, can not support this key file type: %s", err))
+					}
+					passphrase := ""
+					var privateKey interface{}
+					if isEncrypted {
+						if pass, ok := d.GetOk(isBareMetalServerPassphrase); ok {
+							passphrase = pass.(string)
+						} else {
+							return diag.FromErr(fmt.Errorf("Mandatory field 'passphrase' not provided"))
+						}
+						var err error
+						privateKey, err = sshkeys.ParseEncryptedRawPrivateKey(keybytes, []byte(passphrase))
+						if err != nil {
+							return diag.FromErr(fmt.Errorf("Fail to decrypting the private key: %s", err))
+						}
+					} else {
+						var err error
+						privateKey, err = sshkeys.ParseEncryptedRawPrivateKey(keybytes, nil)
+						if err != nil {
+							return diag.FromErr(fmt.Errorf("Fail to decrypting the private key: %s", err))
+						}
+					}
+					var ok bool
+					rsaKey, ok = privateKey.(*rsa.PrivateKey)
+					if !ok {
+						return diag.FromErr(fmt.Errorf("Failed to convert to RSA private key"))
+					}
+				}
+			}
+			// }
+
+			if acc.EncryptedPassword != nil {
+				ciphertext := *acc.EncryptedPassword
+				password := base64.StdEncoding.EncodeToString(ciphertext)
+				if rsaKey != nil {
+					rng := rand.Reader
+					clearPassword, err := rsa.DecryptPKCS1v15(rng, rsaKey, ciphertext)
+					if err != nil {
+						return diag.FromErr(fmt.Errorf("Can not decrypt the password with the given key, %s", err))
+					}
+					password = string(clearPassword)
+				}
+				currAccount[isBareMetalServerUserAccountPassword] = password
```

Comment:
can we set this value only if private_key is provided and we are able to decrypt the password? or mention in the doc saying password will be same as encrypted_password if private_key not provided